java/ql/lib/semmle/code/java/frameworks/android/Intent.qll

```diff
@@ -52,3 +53,126 @@ class BundleGetterMethod extends Method, TaintPreservingCallable {
+        "android.os;Bundle;true;putChar;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putCharArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putCharArray;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequence;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequence;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequenceArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequenceArray;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequenceArrayList;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putCharSequenceArrayList;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putFloat;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putFloatArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putIntegerArrayList;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putParcelable;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putParcelable;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putParcelableArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putParcelableArray;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putParcelableArrayList;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putParcelableArrayList;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putSerializable;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putSerializable;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putShort;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putShortArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putSize;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putSizeF;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putSparceParcelableArray;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putSparseParcelableArray;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;putStringArrayList;;;Argument[0];MapKey of Argument[-1];value",
+        "android.os;Bundle;true;putStringArrayList;;;Argument[1];MapValue of Argument[-1];value",
+        "android.os;Bundle;true;readFromParcel;;;Argument[0];MapKey of Argument[-1];taint",
+        "android.os;Bundle;true;readFromParcel;;;Argument[0];MapValue of Argument[-1];taint",
+        "android.content;Intent;true;getExtras;();;SyntheticField[android.content.Intent.extras] of Argument[-1];ReturnValue;value",
```

Comment:
Since it looks like this deliberately only models the Extras side of Intent, add an inline comment noting this is a partial model